// @syncfusion/ej2-angular-notifications/dist/ej2-angular-notifications.umd.js
/*!
*  filename: ej2-angular-notifications.umd.js
*  version : 18.4.39
*  Copyright Syncfusion Inc. 2001 - 2020. All rights reserved.
*  Use of this code is subject to the terms of our license.
*  A copy of the current license can be obtained at any time by e-mailing
*  licensing@syncfusion.com. Any infringement will be prosecuted under
*  applicable laws. 
*/
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/common'),exports, require('@angular/core'), require('@syncfusion/ej2-angular-base'), require('@syncfusion/ej2-notifications'), require('@angular/common')) :
	typeof define === 'function' && define.amd ? define(['@angular/core','@angular/common','exports', '@angular/core', '@syncfusion/ej2-angular-base', '@syncfusion/ej2-notifications', '@angular/common'], factory) :
	(factory(global.ng.core,global.ng.common,(global['ej2-angular-notifications'] = {}),global.ng.core,global.ej2.angular.base,global.ej2.notifications,global.ng.common));
}(this, (function (ɵngcc0,ɵngcc1,exports,core,ej2AngularBase,ej2Notifications,common) { 
var _c0 = ["title"];
var _c1 = ["content"];
var _c2 = ["template"];
var _c3 = ["*"];
'use strict';

var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var input = ['model'];
var outputs = ['click'];
/**
 * 'e-button' directive represent a button of angular toast
 * It must be contained in a Toast component(`ejs-toast`).
 * ```html
 * <ejs-toast id='toast' showCloseIcon=true>
 *   <e-buttons>
 *    <e-button content='Ok' isPrimary=true></e-button>
 *    <e-button content='Cancel'></e-button>
 *   </e-buttons>
 * </ejs-toast>
 * ```
 */
var ButtonModelPropDirective = /** @class */ (function (_super) {
    __extends(ButtonModelPropDirective, _super);
    /**
     * @param {?} viewContainerRef
     */
    function ButtonModelPropDirective(viewContainerRef) {
        var _this = _super.call(this) || this;
        _this.viewContainerRef = viewContainerRef;
        ej2AngularBase.setValue('currentInstance', _this, _this.viewContainerRef);
        _this.registerEvents(outputs);
        _this.directivePropList = input;
        return _this;
    }
ButtonModelPropDirective.ɵfac = function ButtonModelPropDirective_Factory(t) { return new (t || ButtonModelPropDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef)); };
ButtonModelPropDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ButtonModelPropDirective, selectors: [["e-buttonmodelprop"]], inputs: { model: "model" }, outputs: { click: "click" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ButtonModelPropDirective, [{
        type: core.Directive,
        args: [{
                selector: 'e-buttonmodelprops>e-buttonmodelprop',
                inputs: input,
                outputs: outputs,
                queries: {}
            }]
    }], function () { return [{ type: ɵngcc0.ViewContainerRef }]; }, null); })();
    return ButtonModelPropDirective;
}(ej2AngularBase.ComplexBase));
/**
 * @nocollapse
 */
ButtonModelPropDirective.ctorParameters = function () { return [
    { type: core.ViewContainerRef, },
]; };
/**
 * ButtonModelProp Array Directive
 */
var ButtonModelPropsDirective = /** @class */ (function (_super) {
    __extends(ButtonModelPropsDirective, _super);
    function ButtonModelPropsDirective() {
        return _super.call(this, 'buttons') || this;
    }
ButtonModelPropsDirective.ɵfac = function ButtonModelPropsDirective_Factory(t) { return new (t || ButtonModelPropsDirective)(); };
ButtonModelPropsDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ButtonModelPropsDirective, selectors: [["e-buttonmodelprops"]], contentQueries: function ButtonModelPropsDirective_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, ButtonModelPropDirective, 0);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.children = _t);
    } }, features: [ɵngcc0.ɵɵInheritDefinitionFeature] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ButtonModelPropsDirective, [{
        type: core.Directive,
        args: [{
                selector: 'ejs-toast>e-buttonmodelprops',
                queries: {
                    children: new core.ContentChildren(ButtonModelPropDirective)
                }
            }]
    }], function () { return []; }, null); })();
    return ButtonModelPropsDirective;
}(ej2AngularBase.ArrayBase));
/**
 * @nocollapse
 */
ButtonModelPropsDirective.ctorParameters = function () { return []; };
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
        return Reflect.metadata(k, v);
};
var inputs = ['animation', 'buttons', 'content', 'cssClass', 'enableHtmlSanitizer', 'enablePersistence', 'enableRtl', 'extendedTimeout', 'height', 'icon', 'locale', 'newestOnTop', 'position', 'progressDirection', 'showCloseButton', 'showProgressBar', 'target', 'template', 'timeOut', 'title', 'width'];
var outputs$1 = ['beforeOpen', 'beforeSanitizeHtml', 'click', 'close', 'created', 'destroyed', 'open'];
var twoWays = [''];
/**
 * Represents the Angular Toast Component
 * ```html
 * <ejs-toast></ejs-toast>
 * ```
 */
exports.ToastComponent = /** @class */ (function (_super) {
    __extends(ToastComponent, _super);
    /**
     * @param {?} ngEle
     * @param {?} srenderer
     * @param {?} viewContainerRef
     * @param {?} injector
     */
    function ToastComponent(ngEle, srenderer, viewContainerRef, injector) {
        var _this = _super.call(this) || this;
        _this.ngEle = ngEle;
        _this.srenderer = srenderer;
        _this.viewContainerRef = viewContainerRef;
        _this.injector = injector;
        _this.tags = ['buttons'];
        _this.element = _this.ngEle.nativeElement;
        _this.injectedModules = _this.injectedModules || [];
        _this.registerEvents(outputs$1);
        _this.addTwoWay.call(_this, twoWays);
        ej2AngularBase.setValue('currentInstance', _this, _this.viewContainerRef);
        _this.containerContext = new ej2AngularBase.ComponentBase();
        return _this;
    }
    /**
     * @return {?}
     */
    ToastComponent.prototype.ngOnInit = function () {
        this.containerContext.ngOnInit(this);
    };
    /**
     * @return {?}
     */
    ToastComponent.prototype.ngAfterViewInit = function () {
        this.containerContext.ngAfterViewInit(this);
    };
    /**
     * @return {?}
     */
    ToastComponent.prototype.ngOnDestroy = function () {
        this.containerContext.ngOnDestroy(this);
    };
    /**
     * @return {?}
     */
    ToastComponent.prototype.ngAfterContentChecked = function () {
        this.tagObjects[0].instance = this.childButtons;
        this.containerContext.ngAfterContentChecked(this);
    };
ToastComponent.ɵfac = function ToastComponent_Factory(t) { return new (t || ToastComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Injector)); };
ToastComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ToastComponent, selectors: [["ejs-toast"]], contentQueries: function ToastComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, _c0, 1);
        ɵngcc0.ɵɵcontentQuery(dirIndex, _c1, 1);
        ɵngcc0.ɵɵcontentQuery(dirIndex, _c2, 1);
        ɵngcc0.ɵɵcontentQuery(dirIndex, ButtonModelPropsDirective, 1);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.title = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.content = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.template = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.childButtons = _t.first);
    } }, inputs: { animation: "animation", buttons: "buttons", content: "content", cssClass: "cssClass", enableHtmlSanitizer: "enableHtmlSanitizer", enablePersistence: "enablePersistence", enableRtl: "enableRtl", extendedTimeout: "extendedTimeout", height: "height", icon: "icon", locale: "locale", newestOnTop: "newestOnTop", position: "position", progressDirection: "progressDirection", showCloseButton: "showCloseButton", showProgressBar: "showProgressBar", target: "target", template: "template", timeOut: "timeOut", title: "title", width: "width" }, outputs: { beforeOpen: "beforeOpen", beforeSanitizeHtml: "beforeSanitizeHtml", click: "click", close: "close", created: "created", destroyed: "destroyed", open: "open" }, features: [ɵngcc0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c3, decls: 1, vars: 0, template: function ToastComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2, changeDetection: 0 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ToastComponent, [{
        type: core.Component,
        args: [{
                selector: 'ejs-toast',
                inputs: inputs,
                outputs: outputs$1,
                template: "<ng-content ></ng-content>",
                changeDetection: core.ChangeDetectionStrategy.OnPush,
                queries: {
                    childButtons: new core.ContentChild(ButtonModelPropsDirective)
                }
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ViewContainerRef }, { type: ɵngcc0.Injector }]; }, { title: [{
            type: core.ContentChild,
            args: ['title']
        }], content: [{
            type: core.ContentChild,
            args: ['content']
        }], template: [{
            type: core.ContentChild,
            args: ['template']
        }] }); })();
    return ToastComponent;
}(ej2Notifications.Toast));
/**
 * @nocollapse
 */
exports.ToastComponent.ctorParameters = function () { return [
    { type: core.ElementRef, },
    { type: core.Renderer2, },
    { type: core.ViewContainerRef, },
    { type: core.Injector, },
]; };
exports.ToastComponent.propDecorators = {
    'title': [{ type: core.ContentChild, args: ['title',] },],
    'content': [{ type: core.ContentChild, args: ['content',] },],
    'template': [{ type: core.ContentChild, args: ['template',] },],
};
__decorate([
    ej2AngularBase.Template(),
    __metadata("design:type", Object)
], exports.ToastComponent.prototype, "title", void 0);
__decorate([
    ej2AngularBase.Template(),
    __metadata("design:type", Object)
], exports.ToastComponent.prototype, "content", void 0);
__decorate([
    ej2AngularBase.Template(),
    __metadata("design:type", Object)
], exports.ToastComponent.prototype, "template", void 0);
exports.ToastComponent = __decorate([
    ej2AngularBase.ComponentMixins([ej2AngularBase.ComponentBase]),
    __metadata("design:paramtypes", [core.ElementRef,
        core.Renderer2,
        core.ViewContainerRef,
        core.Injector])
], exports.ToastComponent);
/**
 * NgModule definition for the Toast component.
 */
var ToastModule = /** @class */ (function () {
    function ToastModule() {
    }
ToastModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ToastModule });
ToastModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ToastModule_Factory(t) { return new (t || ToastModule)(); }, imports: [[common.CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ToastModule, { declarations: [
        /**
         * Represents the Angular Toast Component
         * ```html
         * <ejs-toast></ejs-toast>
         * ```
         */
        exports.ToastComponent, ButtonModelPropDirective, ButtonModelPropsDirective], imports: [ɵngcc1.CommonModule], exports: [
        /**
         * Represents the Angular Toast Component
         * ```html
         * <ejs-toast></ejs-toast>
         * ```
         */
        exports.ToastComponent, ButtonModelPropDirective, ButtonModelPropsDirective] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ToastModule, [{
        type: core.NgModule,
        args: [{
                imports: [common.CommonModule],
                declarations: [
                    exports.ToastComponent,
                    ButtonModelPropDirective,
                    ButtonModelPropsDirective
                ],
                exports: [
                    exports.ToastComponent,
                    ButtonModelPropDirective,
                    ButtonModelPropsDirective
                ]
            }]
    }], function () { return []; }, null); })();
    return ToastModule;
}());
/**
 * @nocollapse
 */
ToastModule.ctorParameters = function () { return []; };
/**
 * NgModule definition for the Toast component with providers.
 */
var ToastAllModule = /** @class */ (function () {
    function ToastAllModule() {
    }
ToastAllModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ToastAllModule });
ToastAllModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ToastAllModule_Factory(t) { return new (t || ToastAllModule)(); }, providers: [], imports: [[common.CommonModule, ToastModule], ToastModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ToastAllModule, { imports: [ɵngcc1.CommonModule, ToastModule], exports: [ToastModule] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ToastAllModule, [{
        type: core.NgModule,
        args: [{
                imports: [common.CommonModule, ToastModule],
                exports: [
                    ToastModule
                ],
                providers: []
            }]
    }], function () { return []; }, null); })();
    return ToastAllModule;
}());
/**
 * @nocollapse
 */
ToastAllModule.ctorParameters = function () { return []; };

exports.ButtonModelPropDirective = ButtonModelPropDirective;
exports.ButtonModelPropsDirective = ButtonModelPropsDirective;
exports.ToastModule = ToastModule;
exports.ToastAllModule = ToastAllModule;
exports.ɵa = inputs;
exports.ɵb = outputs$1;
exports.ToastPosition = ej2Notifications.ToastPosition;
exports.ButtonModelProps = ej2Notifications.ButtonModelProps;
exports.ToastAnimations = ej2Notifications.ToastAnimations;
exports.ToastAnimationSettings = ej2Notifications.ToastAnimationSettings;
exports.Toast = ej2Notifications.Toast;

Object.defineProperty(exports, '__esModule', { value: true });

})));
